### PR TITLE
Pull Request for the 5th Milestone of the Rails project Social Milestone 5: Friendships v1

### PR DESCRIPTION
**Pull request by jose-Abel**
Hello,

We're presenting this Pull Request to be reviewed consisting of the fifth milestone for the Rails module, the Social media website project - Milestone 5: Friendships v1. In this milestone, we're requested to implement the Friendship table and model and create it's the relationship with the User table and model. Also we implement the unit test for the models of User, Post, Comment, Like, Friendship and the integration tests for the process of Sign In, Sign Up, Create a Post, Create a Comment and Give a Like.

Hoping to get amazing feedback, happy coding!

### CODE REVIEW

**Review by teekaytech — CHANGES_REQUESTED** [review]
## Status: Changes Required :warning: :vertical_traffic_light: 

### Hello team :wave:

__You've made excellent progress in this milestone with nice implementation of unit and integration test cases :+1: 
However, some changes are required before it can be approved. Please check my comments below and fix them as appropriate.__

**[ REQUIRED CHANGES ]**

_You got a link to this article as a hint: https://smartfunnycool.com/friendships-in-activerecord/. It's a good reference to implement this milestone_

- [x] Add the `.browserslistrc` to .gitignore file and remove it from your repository. This so because the file is neither related not useful to the project 

- [x] User model should have `has_many :inverted_friendships, class_name: "Friendship", foreign_key: "friend_id"` as a relationship ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friend.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))

- [x] User should be able to invite friends ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/send_invitation.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))
- At the moment, a user can only check other user's profiles. The `Send Invitation` feature is not yet implemented.

- [x] User should be able to see friendships requests ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/pending_request.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))
- Again, this feature has not been implemented.

- [x] User has to accept friendship request to start friendship ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/accept_reject_friendship.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))
- Because the friendship feature has not been implemented, this feature is missing too.

- [x] Also, user should be able to reject a friendship request ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/accept_reject_friendship.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))

- [x] There should be a button to invite a user to friendship on the users' list page and on a user page ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/user_can_see_invite_buttton.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))

- [x] Posts timeline should includes posts written by logged-in user and their friends ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))

- [x] Lastly, add more unit and integration tests to cover these features.


* * *
**Please make the required changes and submit another review request when you are done.**

* * *

**Taofeek Olalere
[Twitter](https://twitter.com/ola_lere) | [Github](https://github.com/teekaytech) | @taofeekolalere on Slack**
__Happy Coding :smiley:__

**Review by acushlakoncept — CHANGES_REQUESTED** [review]
# Status: Changes Required ♻️

### Excellent Job, whew! that's a lot, well done in implementing all the changes requested.
Let us quickly add some finishing touches and request another review.

- [x] When I click on the **send invitation** button, it should immediately change to pending invite

- [x] There should be a button to invite a user to friendship on the users' list page 👍  and on the single-user page ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/user_can_see_invite_buttton.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))

- [x] After `user A` sends out an invitation to `user B`, on Logging in with `user B` accounts, I should not see `Send Invitation` again on `user A` profile but `Accept or Reject Friendship` I see you have them reflected on See All Pending Request and See All Friends Request, kindly let the link reflect on the user's list page and user show page as well.

- [x] Sorry, it wasn't pointed out earlier, There should be absolutely NO logic in views. >Views are only for displaying information that became available to it from its corresponding controller not generating it. **Resources**
* [Where Do I Put My Code?](http://codefol.io/posts/Where-Do-I-Put-My-Code)
* [The Beginner's Guide to Rails Helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers)
* [Rails View Helpers Best Practices](https://medium.com/@acushla4real/rails-view-helpers-best-practices-guide-d7d7b6db3db4)
* [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright. See
- `pending_requests.html.erb`, `application.html.erb`,  `_user.html.erb` and `show.html.erb`


### Cheers and Happy coding! 💻 👏

(Feel free to contact me on slack(@acushla), if you have any additional question)
Uduak Essien
[Twitter](https://twitter.com/acushlakoncept) | [LinkedIn](https://www.linkedin.com/in/acushlakoncept/)

**Review by javitocor — APPROVED** [review]
# STATUS APPROVED 🌟 

Good job, team! The project meets all the requirements and it is approved! 🥳 
Congratulations! 💯 
Remember to fill up the completion form 📑 , merge your branch, and start working on the next milestone 🚀 
Good luck with it! 😺

**Pull request merged**
